File: app/src/main/java/com/wegielek/signalychinese/database/RadicalsDao.java
package com.wegielek.signalychinese.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

@Dao
public interface RadicalsDao {

    @Query("SELECT * FROM radicals WHERE section LIKE '%' || :section || '%'")
    LiveData<List<Radicals>> getRadicalsSection(String section);

}

File: app/src/main/java/com/wegielek/signalychinese/database/RadicalsDao.kt
package com.wegielek.signalychinese.database

import androidx.room.Dao
import androidx.room.Query
import com.google.common.util.concurrent.ListenableFuture

@Dao
interface RadicalsDao {
    @Query("SELECT * FROM radicals WHERE section LIKE '%' || :section || '%'")
    fun getRadicalsSection(section: String): ListenableFuture<List<Radicals>>

}